sdk/src/wallet/core/builder.rs

```diff
@@ -178,46 +175,102 @@ where
+            self.address = loaded_address;
+        }
+
+        let loaded_bip_path = loaded_wallet_builder.as_ref().and_then(|builder| builder.bip_path);
 
         // May use a previously stored BIP path if it wasn't provided
         if let Some(bip_path) = self.bip_path {
-            if let Some(restored_bip_path) = restored_bip_path {
-                if bip_path != restored_bip_path {
+            if let Some(loaded_bip_path) = loaded_bip_path {
+                if bip_path != loaded_bip_path {
                     return Err(crate::wallet::Error::BipPathMismatch {
                         new_bip_path: Some(bip_path),
-                        old_bip_path: Some(restored_bip_path),
+                        old_bip_path: Some(loaded_bip_path),
                     });
                 }
+            } else {
+                skip_verification = false;
             }
         } else {
-            self.bip_path = restored_bip_path;
+            self.bip_path = loaded_bip_path;
         }
 
+        // Create the node client.
+        let client = self
+            .client_options
+            .clone()
+            .ok_or(crate::wallet::Error::MissingParameter("client_options"))?
+            .finish()
+            .await?;
+
+        match (self.address.as_ref(), self.bip_path.as_ref()) {
+            (Some(address), Some(bip_path)) => {
+                if !skip_verification {
+                    // verify that the address is derived from the provided bip path.
+                    if let Some(backing_ed25519_address) = address.inner.backing_ed25519() {
+                        self.verify_ed25519_address(backing_ed25519_address, bip_path).await?;
+                    } else {
+                        return Err(crate::wallet::Error::InvalidParameter("address/bip_path mismatch"));
+                    }
+                }
+                self.address.replace(address.clone());
+                self.bip_path = Some(*bip_path);
+            }
+            (Some(address), None) => {
+                self.address.replace(address.clone());
+                self.bip_path = None;
+            }
+            (None, Some(bip_path)) => {
+                if let Some(secret_manager) = &self.secret_manager {
+                    let secret_manager = &*secret_manager.read().await;
+                    let generated_ed25519_address = secret_manager
+                        .generate_ed25519_addresses(
+                            bip_path.coin_type,
+                            bip_path.account,
+                            bip_path.address_index..bip_path.address_index + 1,
+                            GenerateAddressOptions {
+                                internal: bip_path.change != 0,
+                                ledger_nano_prompt: false,
+                            },
+                        )
+                        // Panic: if it didn't return an Err, then there must be at least one address
+                        .await?[0];
```

Comment:
Can we make a function that's generating the address here and that's also called in `verify_ed25519_address()`?